src/main/java/com/arpnetworking/metrics/common/sources/KafkaSource.java

```diff
@@ -138,15 +138,37 @@ private KafkaSource(final Builder<T, V> builder) {
         _parserExecutor = Executors.newFixedThreadPool(_numWorkerThreads);
         _shutdownAwaitTime = builder._shutdownAwaitTime;
         _backoffTime = builder._backoffTime;
+        _bufferSize = builder._bufferSize;
+        _buffer = new ArrayBlockingQueue<>(_bufferSize);
+    }
+
+    // NOTE: Package private for testing
+    /* package private */ KafkaSource(final Builder<T, V> builder, final BlockingQueue<V> buffer) {
```

Comment:
[nit] We have these two ~identical package-private constructors; could we refactor out the commonality into another constructor that they simply delegate to?